src/number_theory/pell.lean

```diff
@@ -137,4 +167,101 @@ end
 
 end existence
 
+/-!
+### Group structure of the solution set
+
+We define a structure of a commutative multiplicative group with distributive negation
+on the set of all solutions to the Pell equation `x^2 - d*y^2 = 1`.
+
+The type of such solutions is `pell.solution₁ d`. It corresponds to a pair of integers `x` and `y`
+and a proof that `(x, y)` is indeed a solution.
+
+The multiplication is given by `(x, y) * (x', y') = (x*y' + d*y*y', x*y' + y*x')`.
+This is obtained by mapping `(x, y)` to `x + y*√d` and multiplying the results.
+In fact, we define `pell.solution₁ d` to be `↥(unitary (ℤ√d))` and transport
+the "commutative group with distributive negation" structure from `↥(unitary (ℤ√d))`.
+
+We then set up an API for `pell.solution₁ d`.
+-/
+
+-- We use `solution₁ d` to allow for a more general structure `solution d m` that
+-- encodes solutions to `x^2 - d*y^2 = m` to be added later.
+
+/-- `pell.solution₁ d` is the type of solutions to the Pell equation `x^2 - d*y^2 = 1`.
+We define this in terms of elements of `ℤ√d` of norm one.
+-/
+@[derive [comm_group, has_distrib_neg, inhabited]]
+def solution₁ (d : ℤ) : Type := ↥(unitary (ℤ√d))
```

Comment:
What does the `₁` refer to? Can you remove it or explain it in the docstring?

Reply:
It could be left out. I put it in to make clear that `unitary ℤ√d` is converted to a type (it is a `submonoid` a priori).

Reply:
I think it's nice to have it

Reply:
Oh okay that's not at all what I was expecting. I thought it referred to the fact that those are solutions to `x^2 - d * y ^2 = 1` as opposed to the more general `x^2 - d * y ^2 = m`.

How is `₁` referring to `Type` instead of `submonoid`?

Reply:
Maybe @MichaelStollBayreuth misread and thought you were asking about the `↥`?

Reply:
Sorry, I misread your `₁` for `↥`. Yes, the subscript is there to indicate that the norm is 1 and not something else.
I'm thinking of eventually adding a proof of the statement that the solution set of `x^2 - d*y^2 = m` is a finite union of orbits under the natural action of `solution₁ d`.

Reply:
@tb65536 You beat me to it! :smile: